### PR TITLE
feat: vendor GPv2TradeEncoder

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes BAL-103

### CODE REVIEW

**Review by 0xteddybear — APPROVED** [review]
question: why use `main^` instead of `main` commit of composable-cow?  did the last commit break anything for us?

nitpick: I would have committed separately
- the openzeppelin remapping + related changes
- the cowswap remapping + related changes
- using GPv2TradeEncoder from imported dependency

**Comment by 0xAustrian** [comment]
> question: why use `main^` instead of `main` commit of composable-cow? did the last commit break anything for us?
> 
> nitpick: I would have committed separately
> 
> * the openzeppelin remapping + related changes
> * the cowswap remapping + related changes
> * using GPv2TradeEncoder from imported dependency

oh, I had the hash of the commit saved but didn't noticed a new commit. And I think this new commit will be better. Let me fix it.

**Review by wei3erHase — DISMISSED** [review]
LGTM 🚀

**Review by wei3erHase — APPROVED** [review]
a-ma-zing! 👏

File: remappings.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ ds-test/=node_modules/ds-test/src
 forge-std/=node_modules/forge-std/src
 forge-gas-snapshot/=node_modules/forge-gas-snapshot/src
 solmate/=node_modules/solmate/src
+@cowprotocol/=node_modules/@cowprotocol/contracts/src/contracts
 cowprotocol/=node_modules/@cowprotocol/contracts/src/
 composable-cow/=node_modules/composable-cow/
```
Comment (wei3erHase) [inline remappings.txt:7]: comment, wdyt about using `@composable-cow` and adding a comment on why are we declaring the `cowprotocol` remapping, that is internal to their codebase

**Review by wei3erHase — APPROVED** [review]

**Review by 0xteddybear — APPROVED** [review]

**Pull request merged**